src/rust/mqtt-publisher/src/main.rs

```diff
@@ -0,0 +1,255 @@
+                        false,
+                        serde_json::to_string(&data)?,
+                    )
+                    .await?;
+            }
+            "place_market_order" => {
+                let data: PlaceMarketOrderNotif = serde_json::from_str(notification.payload())?;
+                client
+                    .publish(
+                        format!(
+                            "place_market_order/{}/{}/{}/{}",
+                            data.market_id, data.user, data.custodian_id, data.integrator
+                        ),
+                        QoS::AtLeastOnce,
+                        false,
+                        serde_json::to_string(&data)?,
+                    )
+                    .await?;
+            }
+            "place_swap_order" => {
+                let data: PlaceSwapOrderNotif = serde_json::from_str(notification.payload())?;
+                client
+                    .publish(
+                        format!(
+                            "place_swap_order/{}/{}/{}",
+                            data.market_id, data.integrator, data.signing_account
+                        ),
+                        QoS::AtLeastOnce,
+                        false,
+                        serde_json::to_string(&data)?,
+                    )
+                    .await?;
+            }
+            "change_order_size" => {
+                let data: ChangeOrderSizeNotif = serde_json::from_str(notification.payload())?;
+                client
+                    .publish(
+                        format!(
+                            "change_order_size/{}/{}/{}",
+                            data.market_id, data.user, data.custodian_id
+                        ),
+                        QoS::AtLeastOnce,
+                        false,
+                        serde_json::to_string(&data)?,
+                    )
+                    .await?;
+            }
+            "cancel_order" => {
+                let data: CancelOrderNotif = serde_json::from_str(notification.payload())?;
+                client
+                    .publish(
+                        format!(
+                            "cancel_order/{}/{}/{}",
+                            data.market_id, data.user, data.custodian_id
+                        ),
+                        QoS::AtLeastOnce,
+                        false,
+                        serde_json::to_string(&data)?,
+                    )
+                    .await?;
+            }
+            "fill" => {
+                let data: FillNotif = serde_json::from_str(notification.payload())?;
+                if !emitted_fills.remove(&(
+                    data.market_id.clone(),
+                    data.maker_address.clone(),
+                    data.maker_custodian_id.clone(),
+                    data.taker_address.clone(),
+                    data.taker_custodian_id.clone(),
+                    data.sequence_number_for_trade.clone(),
```

Comment:
For deduping, all you need is:

- [ ] `market_id`
- [ ] `taker_order_id`
- [ ] `sequence_number_for_trade`